src/components/ButtonOption/__test__/buttonOption.spec.js

```diff
@@ -0,0 +1,100 @@
+import React from 'react';
+import { mount } from 'enzyme';
+import { Component as ButtonGroupPicker } from '../../ButtonGroupPicker/index';
+import ButtonOption from '../index';
+
+jest.mock('../helpers/isOptionSelected', () => jest.fn(() => true));
+
+describe('<ButtonOption />', () => {
+    it('should set disabled prop of input to true when disabled prop is true', () => {
+        const component = mount(
+            <ButtonGroupPicker>
+                <ButtonOption label="Option 1" disabled />
+            </ButtonGroupPicker>,
+        );
+        expect(component.find('input').prop('disabled')).toBe(true);
+    });
+
+    it('should set the label passed as label for the input', () => {
+        const component = mount(
+            <ButtonGroupPicker>
+                <ButtonOption label="Option 1" />
+            </ButtonGroupPicker>,
+        );
+        expect(component.find('label').text()).toBe('Option 1');
+    });
+
+    it('should fire onclick callback when clicked', () => {
+        const onClickFn = jest.fn();
+        const component = mount(
+            <ButtonGroupPicker>
+                <ButtonOption label="Option 1" onClick={onClickFn} />
+            </ButtonGroupPicker>,
+        );
+        component.find('label').simulate('click');
+        expect(onClickFn).toHaveBeenCalledTimes(1);
+    });
+
+    it('should pass a generated id to the input element and set the same id to the htmFor of the label element', () => {
+        const component = mount(
+            <ButtonGroupPicker>
+                <ButtonOption label="Option 1" />
+            </ButtonGroupPicker>,
+        );
+        const htmlFor = component.find('label').prop('htmlFor');
+        const inputId = component.find('input').prop('id');
+        expect(htmlFor).toBe(inputId);
+    });
+
+    it('should set the name passed through context to the name prop of the input element', () => {
+        const component = mount(
+            <ButtonGroupPicker>
+                <ButtonOption label="Option 1" />
+            </ButtonGroupPicker>,
+        );
+        const contextName = component.instance().getContext().name;
```

Comment:
same ^^ let's avoid the use of component.instance() and instead get this value from value prop in Provider component